sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/challenge_auth_policy.py

```diff
@@ -54,28 +56,56 @@ def send(self, request):
             )
             if request.http_request.body:
                 # no_body was created with request's headers -> if request has a body, no_body's content-length is wrong
+                _LOGGER.debug("no_body was created with request's headers."
+                              "If request has a body, no_body's content length is wrong")
                 no_body.headers["Content-Length"] = "0"
-
+            _LOGGER.info("Provoking challenge with an unauthorized, bodiless {} to url {} with the following headers".
+                         format(request.http_request.method, request.http_request.url))
+            for header in request.http_request.headers:
+                _LOGGER.info("Header {}".format(header))
```

Comment:
This information is not logged in the async policy

Reply:
Changed the format of both this one and the async one so there isn't a separate line for each header.